### PR TITLE
Creates an endpoint to create user activities and make an initial schedule of treats

### LINKED ISSUE #36: Integrate Google Calendar API into Service Class
**Chore**
Use playground code to setup Google Service Class
- [x] make a method to initiate service
- [x] create testing

### PR DESCRIPTION
**Pull request by rer7891**
### Sidebar Checklist

- [x] Request reviewers
- [x] Assign yourself and other contributors
- [x] Link to project

### Issues Resolved
Resolves #36 #37 #38 

### Problem Addressed
Need an endpoint for the Front End to hit that passes in a user's preferred Categories then assigns associated activities to that user. Will also return a list of scheduled "treats" that are randomly selected based on a user's calendar availability and selected categories.
### Solution Implemented
Creates an endpoint that uses a scheduler service class to call on the Google Calendar API. 
### Other Notes
If you want to test this locally you will need to replace the authorization in the Scheduler Service from current_user.google_token to a personal google token. Google Tokens can be generated here: 
https://developers.google.com/oauthplayground/  using the Calendar/auth setting.

### CODE REVIEW

File: app/poros/suggestion_facade.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class SuggestionFacade
+	def initialize(current_user)
+		@current_user = current_user
+	end
+
+	def schedule_activities
+		scheduler = SchedulerService.new(@current_user)
```
Comment (tyladevon) [inline app/poros/suggestion_facade.rb:7]: Good abstraction here pulling our service into ScheduleService from the Suggestions Facade

File: app/services/scheduler_service.rb [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  def find_user_events
+    calendar_id = "primary"
+    response = get_calendar_service.list_events( calendar_id,
+                                   time_min: DateTime.now.rfc3339,
+                                   time_max: (DateTime.now + 1.week).rfc3339,
+                                   single_events: true,
+                                   order_by: "startTime" )
+  end
+
+  # Sets available times from 8 a.m. - 8 p.m. as a string and puts that
```
Comment (tyladevon) [inline app/services/scheduler_service.rb:34]: Thank you for adding in the comments in this file. Will be worth having everyone read through it before Wednesday and then possibly removing it?

File: app/services/scheduler_service.rb [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    available_times.delete_if do |time|
+    # since this is a string this deletes any minute over 00:59 ex. (09:60 would be deleted)
+    # this returns an array with more realistic time frames
+     time.split("")[3].to_i > 5
+    end
+  end
+
+  # Uses the array of events to subract event time frames from a user's availability.
+  def availability
+    find_user_events.items.reduce({}) do |acc, event|
```
Comment (tyladevon) [inline app/services/scheduler_service.rb:47]: Nice use of reduce enum here. Dries up some of the potentially long lines of logic.

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -18,8 +18,7 @@
   config.cassette_library_dir = 'spec/cassettes'
   config.hook_into :webmock
   config.configure_rspec_metadata!
-  config.filter_sensitive_data("") { ENV[''] }
-  config.filter_sensitive_data("") { ENV[''] }
+  config.filter_sensitive_data("GOOGLE_TOKEN") { ENV['GOOGLE_TOKEN'] }
```
Comment (tyladevon) [inline spec/rails_helper.rb:21]: Thanks for remembering to filter out tokens here.

File: spec/requests/api/v1/users_spec.rb [hunk 5]
```diff
@@ -11,29 +11,11 @@
 
 		post '/api/v1/users', params: params
 
-		post '/graphql', params: { query: query }
-
 		expect(response).to be_successful
-		JSON.parse(response.body, symbolize_names: true)
-	end
+		expect(User.count).to eq(1)
+		user = User.first
```
Comment (tyladevon) [inline spec/requests/api/v1/users_spec.rb:16]: Thank you for making this test more robust than what I pushed up!

File: app/services/scheduler_service.rb (hunk 2 above)
Comment (wrenisit) [reply app/services/scheduler_service.rb:34]: This is a great solution. Thank you for the comments to help with understanding the code.

File: app/services/scheduler_service.rb [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+      # If this key has not been assigned then sets the availability to full availability
+        available_times = available_time
+      else
+      # If this key has been assigned it sets available times to the existing value
+        available_times = acc[weekday]
+      end
+      # Sets key to weekday and the value to available time minus the event time.
+      acc[weekday] = [available_times - event_range].flatten
+      acc
+    end
```
Comment (wrenisit) [inline app/services/scheduler_service.rb:72]: Great covering of contingencies here.

**Pull request merged**